graphql_compiler/compiler/emit_cypher.py

```diff
@@ -127,10 +118,130 @@ def _emit_with_clause_components(cypher_steps):
     return result
 
 
+def _emit_with_clause_components_for_current_fold_scope(current_fold_scope_cypher_steps):
+    """Emit a list of strings, one for each vertex or list passing through a WITH clause."""
+    result = []
+
+    vertex_names = {}
+    for cypher_step in current_fold_scope_cypher_steps:
+        if not(isinstance(cypher_step, CypherStep)):
+            raise TypeError(u'Expected current_fold_scope_cypher_steps to contain only CypherStep '
+                            u'objects. Instead, got object {} of type {}. '
+                            u'current_fold_scope_cypher_steps: {}'
+                            .format(cypher_step, type(cypher_step),
+                                    current_fold_scope_cypher_steps))
+        fold_scope_location = cypher_step.as_block.location
+        full_vertex_name = cypher_helpers.get_fold_scope_location_full_path_name(
+            fold_scope_location)
+        collected_name = cypher_helpers.get_collected_vertex_list_name(full_vertex_name)
+        vertex_names[u'collect(' + full_vertex_name + ')'] = collected_name
+
+    # Sort the locations, to ensure a deterministic order.
+    for index, collect_call in enumerate(sorted(vertex_names)):
+        if index > 0:
+            result.append(u',')
+        result.append(u'\n  %(collect_call)s AS %(collected_name)s' %
```

Comment:
minor nit: prefer `.format()` instead of `%` formatting unless there's a specific reason not to do that. MATCH uses `%` formatting because curly braces are everywhere and `.format()` gets really annoying when you have to escape all those braces. Here it doesn't seem there's a strong reason to use `%` so I'd go with `.format`